git/git.go

```diff
@@ -385,7 +393,7 @@ func (provider *repositoryProvider) snapshot(repository *git.Repository, commit
 
 			err = addEntryToIndexFile(indexOutputFile, name, &entry)
 			if err != nil {
-				break
+				return 0, fmt.Errorf("failed to write to index file for '%v': %v", name, err)
```

Comment:
you missed one break above

Reply:
OHhh ok, i found the issue here with all the breaks, there was that there's an additional 'err' variable thats hiding the outer scope